torch/utils/_config_module.py

```diff
@@ -157,13 +193,18 @@ def __getattr__(self, name: str) -> Any:
             if config.user_override is not _UNSET_SENTINEL:
                 return config.user_override
 
+            if config.justknob is not None:
+                # JK only supports bools and ints
+                return justknobs_check(name=config.justknob, default=config.default)
+
             # Note that reference types can still be modified, so we
             # copy them to user_overrides in case the user overrides
             # them
             if isinstance(config.default, (list, set, dict)):
                 config.user_override = copy.deepcopy(config.default)
                 return config.user_override
```

Comment:
Something I realized rereading this is, I think we need to ban JKs and other sidechannel influencer things on mutable data types. The problem is you are transparently initializing the `user_override` here. But that means that something like JK will *no longer apply* (because user override has precedence over JK), even if the JK later gets swapped. Now, they are de facto banned because you can't actually put structured types in JK..... but IDK, this is just a very annoying special case lol

Reply:
I am almost tempted to say that we should return a special "mutable view" of the config, which only issues a user override when a mutation occurs. Or, I would like to simply ban mutable types in the config. Well, that's for another time...

Reply:
Let me stack a PR to at least ban mutable types, that's relatively "straightforward".

Reply:
https://github.com/pytorch/pytorch/pull/139488